Test failures after upgrading Django app to mysqlclient 1.3.14

Hi 

### STR:
1. Have a Django 1.11 app whose tests make use of [CaptureQueriesContext](https://github.com/django/django/blob/1.11.17/django/test/utils.py#L621-L654), such as:
    - https://github.com/mozilla/treeherder/blob/d36bb8ab7644d3afe4b21e2b283c77e51aca1a15/tests/model/test_bugscache.py#L151-L179
2. Upgrade the `mysqlclient` used by that app from 1.3.13 to 1.3.14, eg:
    - mozilla/treeherder/pull/4342
3. Run test suite (either locally or on Travis)

### Expected:
No test failures after upgrading, since:
- it was a minor release
- there was no mention of removing `_last_executed` in the changelog

### Actual:
Test failures, eg:

```
        with CaptureQueriesContext(connection) as context:
            Bugscache.search(search_term)
    
            search_query = get_search_query(context.captured_queries[0]['sql'])
>           assert search_query
E           AssertionError: assert ''
```
(https://travis-ci.org/mozilla/treeherder/jobs/463284994#L605)

Whilst Django's handling has been updated for Django 2.1+, this leaves Django 1.11 (LTS) and Django 2.0 incompatible with latest mysqlclient.

If restoring `_last_executed` ends up being wontfix, please can the changelog be updated to mention the removal (and if possible a reference to say Django <2.1 users should not update)?

Many thanks :-)

Refs:
- https://github.com/PyMySQL/mysqlclient-python/pull/283#issuecomment-444562762
- https://code.djangoproject.com/ticket/30013
- django/django/pull/10726
- django/django/pull/10727

--- a/MySQLdb/connections.py
+++ b/MySQLdb/connections.py
@@ -166,6 +166,14 @@ class Connection(_mysql.connection):
         self.encoders = dict([ (k, v) for k, v in conv.items()
                                if type(k) is not int ])
 
+        # XXX THIS IS GARBAGE: While this is just a garbage and undocumented,
+        # Django 1.11 depends on it.  And they don't fix it because
+        # they are in security-only fix mode.
+        # So keep this garbage for now.  This will be removed in 1.5.
+        # See PyMySQL/mysqlclient-python#306
+        self.encoders[unicode] = unicode
+        self.encoders[bytes] = bytes
+
         self._server_version = tuple([ numeric_part(n) for n in self.get_server_info().split('.')[:2] ])
 
         self.encoding = 'ascii'  # overridden in set_character_set()

--- a/MySQLdb/connections.py
+++ b/MySQLdb/connections.py
@@ -171,7 +171,6 @@ class Connection(_mysql.connection):
         # they are in security-only fix mode.
         # So keep this garbage for now.  This will be removed in 1.5.
         # See PyMySQL/mysqlclient-python#306
-        self.encoders[unicode] = unicode
         self.encoders[bytes] = bytes
 
         self._server_version = tuple([ numeric_part(n) for n in self.get_server_info().split('.')[:2] ])

--- a/MySQLdb/connections.py
+++ b/MySQLdb/connections.py
@@ -245,8 +245,11 @@ class Connection(_mysql.connection):
             s = self.string_literal(o.encode(self.encoding))
         elif isinstance(o, bytearray):
             s = self._bytes_literal(o)
-        elif not PY2 and isinstance(o, bytes):
-            s = self._bytes_literal(o)
+        elif isinstance(o, bytes):
+            if PY2:
+                s = self.string_literal(o)
+            else:
+                s = self._bytes_literal(o)
         elif isinstance(o, (tuple, list)):
             s = self._tuple_literal(o)
         else:

--- a/MySQLdb/cursors.py
+++ b/MySQLdb/cursors.py
@@ -63,6 +63,14 @@ class BaseCursor(object):
         self.rowcount = -1
         self.arraysize = 1
         self._executed = None
+
+        # XXX THIS IS GARBAGE: While this is totally garbage and private,
+        # Django 1.11 depends on it.  And they don't fix it because
+        # they are in security-only fix mode.
+        # So keep this garbage for now.  This will be removed in 1.5.
+        # See PyMySQL/mysqlclient-python#303
+        self._last_executed = None
+
         self.lastrowid = None
         self.messages = []
         self._result = None
@@ -300,6 +308,7 @@ class BaseCursor(object):
         self._do_get_result(db)
         self._post_get_result()
         self._executed = q
+        self._last_executed = q  # XXX THIS IS GARBAGE: See above.
         return self.rowcount
 
     def _fetch_row(self, size=1):

